src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/SingleThreadedDispatcher.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    internal abstract class SingleThreadedDispatcher
+    {
+        public abstract bool IsDispatcherThread { get; }
```

Comment:
Do we still need to expose this and the `DispatcherScheduler` properties?

Reply:
Good catch, we don't need to expose `IsDispatcherThread` anymore so I made it protected. We do need to still expose `DispatcherScheduler` because it's used in a few places (e.g. a `ContinueWith` statements as well as some test scenarios where I don't think there's an alternative).